packaging/test_irods_client_rest_cpp.py

```diff
@@ -350,6 +350,34 @@ def test_list_with_accoutrements(self):
                 os.remove(file_name)
                 admin.assert_icommand(['irm', '-f', file_name])
 
+    def test_query_handles_case_insensitivity__issue124(self):
+        with session.make_session_for_existing_admin() as admin:
+            try:
+                case_sensitive = 0
+
+                dobj_name = 'mAy_I_tAkE_yOuR_hAt_SIR'
+                a_path = os.path.join(admin.home_collection, dobj_name)
+                admin.assert_icommand(['itouch', a_path])
+
+                token  = irods_rest.authenticate('rods', 'rods', 'native')
+
+                query_lower = "SELECT DATA_PATH, COLL_NAME, DATA_NAME, DATA_SIZE WHERE DATA_NAME LIKE '%i%'"
+                query_upper = "SELECT DATA_PATH, COLL_NAME, DATA_NAME, DATA_SIZE WHERE DATA_NAME LIKE '%I%'"
+
+                result_lower = irods_rest.query(token, query_lower, 1, 0, 'general')
+                result_upper = irods_rest.query(token, query_upper, 1, 0, 'general', _case_sensitive=case_sensitive)
```

Comment:
pretty sure the assert will fail now.

i think you want the case_sensitive=0 on query_lower instead...